src/server/shared/Updater/DBUpdater.cpp

```diff
@@ -349,7 +349,31 @@ void DBUpdater<T>::ApplyFile(DatabaseWorkerPool<T>& pool, std::string const& hos
     args.push_back("-h" + host);
     args.push_back("-u" + user);
     args.push_back("-p" + password);
-    args.push_back("-P" + port_or_socket);
+
+    // Check If first character is numeric
+    bool _uses_Socket = false;
+    switch (port_or_socket[0])
+    {
+        case '0':case '1':case '2':case '3':case '4':
+        case '5':case '6':case '7':case '8':case '9': break;
+        default:
+            _uses_Socket = true;
+            break;
+    }
```

Comment:
I got the full one with support for windows and linux merged on this pull request.

Thanks @DDuarte for point me some things